tests/lib/k8s/test_builders.py

```diff
@@ -277,22 +304,281 @@ def test_get_secret_variables(self):
+            {"metadata": {"name": "deployment"}},
+        ]
+
+        namespace = k8s_builders.set_namespace(
+            app_config, resources, self.compiled_templates_mock()
+        )
+
+        self.assertEqual(namespace, None)
+        self.assertEqual(
+            resources, [{"metadata": {"name": "hpa"}}, {"metadata": {"name": "deployment"}},]
+        )
+
+    @patch("yaml_lib.parse_yaml", autospec=True)
+    def test_set_namespace_with_namespace(self, parse_yaml_mock):
+        """Return namespace and should modify resources definition"""
+        parse_yaml_mock.return_value = k8s_fixtures.namespace
+
+        app_config = {"namespace": "namespace"}
+        resources = [
+            {"metadata": {"name": "hpa"}},
+            {"metadata": {"name": "deployment"}},
+        ]
+
+        namespace = k8s_builders.set_namespace(
+            app_config, resources, self.compiled_templates_mock()
+        )
+
+        self.assertEqual(namespace, k8s_fixtures.namespace)
+        self.assertEqual(
+            resources,
+            [
+                {"metadata": {"name": "hpa", "namespace": "namespace"}},
+                {"metadata": {"name": "deployment", "namespace": "namespace"}},
+            ],
+        )
+
+    def test_set_replicas_min_replicas_zero(self):
+        """Return None and set replicas to 0 in the recipe"""
+        app_config = {"app": "app", "scales": {"web": {"minReplicas": 0}}}
+        recipe = {"spec": {}}
+        hpa = k8s_builders.set_replicas(app_config, "web", recipe, self.compiled_templates_mock())
+        self.assertEqual(hpa, None)
+        self.assertEqual(recipe, {"spec": {"replicas": 0}})
+
+    @patch("yaml_lib.parse_yaml", autospec=True)
+    def test_set_replicas_with_process_default(self, parse_yaml_mock):
+        """Return the hpa configuration"""
+        parse_yaml_mock.return_value = k8s_fixtures.hpa
+
+        app_config = {
+            "app": "app",
+            "scales": {
+                "default": {
+                    "minReplicas": 5,
+                    "maxReplicas": 15,
+                    "targetCPUUtilizationPercentage": 90,
+                },
+            },
+        }
+        recipe = {"spec": {}}
+
+        hpa = k8s_builders.set_replicas(app_config, "web", recipe, self.compiled_templates_mock())
+
+        self.assertEqual(hpa, k8s_fixtures.hpa)
+
+    @patch("yaml_lib.parse_yaml", autospec=True)
+    def test_set_replicas_with_process_configuration(self, parse_yaml_mock):
+        """Return the hpa configuration"""
+        parse_yaml_mock.return_value = k8s_fixtures.hpa
+
+        app_config = {
+            "app": "app",
+            "scales": {
+                "default": {
+                    "minReplicas": 3,
+                    "maxReplicas": 9,
+                    "targetCPUUtilizationPercentage": 80,
+                },
+                "web": {"minReplicas": 5, "maxReplicas": 15, "targetCPUUtilizationPercentage": 90,},
```

Comment:
It would be nice if `black` automatically cleaned those
```suggestion
                "web": {"minReplicas": 5, "maxReplicas": 15, "targetCPUUtilizationPercentage": 90},
```